src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc

```diff
@@ -292,17 +300,40 @@ void ResourceAllocationManagerImpl::SetResourceAquired(
     const std::string& module_type, const uint32_t app_id) {
   LOG4CXX_AUTO_TRACE(logger_);
   allocated_resources_[module_type] = app_id;
-  SendOnRCStatusNotification();
+  SendOnRCStatusNotifications(NotificationTrigger::MODULE_ALLOCATION);
 }
 
-void ResourceAllocationManagerImpl::SendOnRCStatusNotification() {
+void ResourceAllocationManagerImpl::SendOnRCStatusNotifications(
+    NotificationTrigger::eType event) {
   LOG4CXX_AUTO_TRACE(logger_);
   auto rc_apps = RCRPCPlugin::GetRCApplications(app_mngr_);
   for (const auto& rc_app : rc_apps) {
-    auto msg_to_mobile = CreateOnRCStatusNotificationToMobile(rc_app);
-    rpc_service_.SendMessageToMobile(msg_to_mobile);
-    auto msg_to_hmi = CreateOnRCStatusNotificationToHmi(rc_app);
-    rpc_service_.SendMessageToHMI(msg_to_hmi);
+    smart_objects::SmartObjectSPtr msg_to_mobile;
+    smart_objects::SmartObjectSPtr msg_to_hmi;
+    switch (event) {
+      case NotificationTrigger::APP_REGISTRATION:
+        msg_to_mobile = CreateOnRCStatusNotificationToMobile(rc_app);
+        (*msg_to_mobile)[application_manager::strings::msg_params]
+                        [message_params::kAllowed] = is_rc_enabled();
+        rpc_service_.SendMessageToMobile(msg_to_mobile);
+        if (is_rc_enabled()) {
```

Comment:
why send notification to HMI when an app registers?

Reply:
never mind, now I understand it is the first notification tell the HMI that the app has no allocated modules.

Reply:
notification shall be sent to newly registered RC app, not to existing RC apps, as there is no change to the two lists